app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -33,11 +33,13 @@ def command_headers
     }
 
     if registration_params['jwt_expiration'].present?
-      jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i if registration_params['jwt_expiration'] != 'unlimited'
+      jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i if registration_params['jwt_expiration'] != 'unlimited' && registration_params['jwt_expiration'].to_i != -1
```

Comment:
This is not readable much, maybe the `case`  statement here would be better.
Also why to have the condition for the `unlimited`, that should be changed to `-1` right?

Reply:
The value of `jwt_args[:expiration]` must be nil when the token value is 'unlimited' (from the UI) or -1 from the api or hammer so this line is basically not assigning anything to  `jwt_args[:expiration]` when the either of them is the case. 
We can also convert 'unlimited' (coming from the UI) to -1 and then send nil to  `jwt_args[:expiration]` when the case is -1 but that's just an extra step. Whatever works. 
I agree with the case statements. I'll update that.